core/common/src/main/java/alluxio/conf/PropertyKey.java

```diff
@@ -5279,6 +5325,12 @@ private static String javadocLink(String fullyQualifiedClassname) {
         "alluxio.worker.block.master.client.pool.size";
     public static final String WORKER_PRINCIPAL = "alluxio.worker.principal";
     public static final String WORKER_RAMDISK_SIZE = "alluxio.worker.ramdisk.size";
+    public static final String WORKER_REVIEWER_BUFFER_HARDLIMIT_BYTES =
+            "alluxio.worker.reviewer.buffer.hardlimit.bytes";
```

Comment:
Is the namespace for this property correct? Since it seems to only affect the probabilistic reviewer.

Probably not something we need to address in this PR - if users are plugging in their own reviewers, how will they add configuration options?

Reply:
By the `buffer` i mean the `ProbabilisticBufferReviewer`, but I see it can be confusing due to the multiple meanings of `buffer`. On second thought I'm changing it to `alluxio.worker.reviewer.probabilistic.hardlimit.bytes` to make the reference to `ProbabilisticBufferReviewer` clearer.

For plugging in own Reviewer implementation, that would be `alluxio.worker.reviewer.<foo>.<property key>`. Does that make more sense to you?

Reply:
Ah okay, I didn't make the connection, this is much clearer. My question about plugins is more around the fact that they would need to add the property key to our `PropertyKey.java` file which requires recompiling.

Reply:
I see, that's a very good question. I did a random search for our policy configuration like `Allocator`, `Evictor` and `BlockLocationPolicy`. The implementations seem to be using hard coded parameters so I don't find an existing great way of plugging in customized properties w/o recompiling. If you are not aware of an existing solution either, I think we'll need to define how that's done.